sw/device/tests/hmac_dif_test.c

```diff
@@ -148,143 +159,95 @@ static bool check_message_length(const dif_hmac_t *hmac,
+      case kDifHmacDigestOk:
         break;
-      case kDifHmacBadArg:
-        LOG_ERROR(
-            "Invalid arguments encountered checking HMAC run status.\r\n");
+      case kDifHmacDigestBadArg:
+        LOG_ERROR("Invalid arguments encountered reading HMAC digest.");
         return false;
+      case kDifHmacDigestProcessing:
+        hmac_done = false;
       default:
-        LOG_ERROR("Unknown error encountered checking HMAC run status.\r\n");
+        LOG_ERROR("Unknown error encountered reading HMAC digest.");
         return false;
     }
   } while (!hmac_done);
 
-  return true;
-}
-
-/**
- * Read the HMAC digest and compare it to the expected result. Return |true| if
- * there are no errors and the digests match.
- */
-static bool check_digest(const dif_hmac_t *hmac,
-                         const uint32_t *expected_digest) {
-  dif_hmac_digest_t digest_result;
-  switch (dif_hmac_digest_read(hmac, &digest_result)) {
-    case kDifHmacOk:
-      break;
-    case kDifHmacBadArg:
-      LOG_ERROR("Invalid arguments encountered reading HMAC digest.\r\n");
-      return false;
-    default:
-      LOG_ERROR("Unknown error encountered reading HMAC digest.\r\n");
-      return false;
-  }
-
   bool matches = true;
   for (int i = 0; i < 8; ++i) {
     if (expected_digest[i] != digest_result.digest[i]) {
-      LOG_ERROR(
-          "Digest mismatch, expected: 0x%X at index [%i] but got: "
-          "0x%X.\r\n",
-          expected_digest[i], i, digest_result.digest[i]);
+      LOG_ERROR("Digest mismatch, expected: 0x%X at index [%i] but got: 0x%X.",
+                expected_digest[i], i, digest_result.digest[i]);
       matches = false;
     }
   }
 
   return matches;
 }
 
-static bool run_test_impl(const dif_hmac_config_t *config, const char *data,
-                          size_t len, const uint32_t *expected_digest) {
+static void run_test(const dif_hmac_config_t *config, const char *data,
+                     size_t len, const uint8_t *key,
+                     const uint32_t *expected_digest) {
   dif_hmac_t hmac;
-  if (!test_setup(config, &hmac)) {
-    return false;
-  }
-
-  if (!push_message(&hmac, data, len)) {
-    return false;
-  }
-
-  if (!wait_for_fifo_empty(&hmac)) {
-    return false;
-  }
 
-  if (!check_message_length(&hmac, len * 8)) {
-    return false;
-  }
-
-  if (!run_hmac(&hmac)) {
-    return false;
-  }
-
-  if (!wait_for_hmac_done(&hmac)) {
-    return false;
-  }
-
-  if (!check_digest(&hmac, expected_digest)) {
-    return false;
-  }
-
-  return true;
-}
-
-void run_test(char *test_name, const dif_hmac_config_t *config,
-              const char *data, size_t len, const uint32_t *expected_digest) {
-  LOG_INFO("Running test \"");
-  uart_send_str(test_name);
-  uart_send_str("\":\r\n");
-  if (run_test_impl(config, data, len, expected_digest)) {
-    LOG_INFO("PASS!\r\n");
-  } else {
-    LOG_INFO("FAIL!\r\n");
-  }
+  CHECK(test_setup(config, &hmac, key));
```

Comment:
If you use `CHECK` inside the lower level functions, you won't need to check it here. `CHECK` does `abort` internally, it would then simplify things slightly.